internal/kafka/internal/workers/clusters_mgr.go

```diff
@@ -518,6 +543,39 @@ func (c *ClusterManager) reconcileEmptyCluster(cluster api.Cluster) (bool, error
 	return updateStatusErr == nil, updateStatusErr
 }
 
+func (c *ClusterManager) reconcileWaitingForKasFleetshardOperatorCluster(cluster api.Cluster) error {
+	glog.V(10).Infof("check if cluster is empty, ClusterID = %s", cluster.ClusterID)
+	var err error
+
+	if err := c.reconcileClusterResources(cluster); err != nil {
+		return errors.WithMessagef(err, "failed to reconcile  waiting for Kas Fleetshard Operator cluster resources '%s'", cluster.ClusterID)
+	}
+
+	err = c.reconcileClusterIdentityProvider(cluster)
+	if err != nil {
+		return errors.WithMessagef(err, "failed to reconcile identity provider of waiting for Kas Fleetshard Operator cluster %s: %s", cluster.ClusterID, err.Error())
+	}
+
+	if c.KasFleetshardOperatorAddon != nil {
+		if params, err := c.KasFleetshardOperatorAddon.ReconcileParameters(cluster); err != nil {
+			if err.IsBadRequest() {
+				glog.Infof("kas-fleetshard operator is not found on cluster %s", cluster.ClusterID)
+			} else {
+				return errors.WithMessagef(err, "failed to reconcile kas-fleet-shard parameters of waiting for Kas Fleetshard Operator cluster %s: %s", cluster.ClusterID, err.Error())
+			}
+		} else {
+			if cluster.ClientID == "" {
+				cluster.ClientID = params.GetParam(services.KasFleetshardOperatorParamServiceAccountId)
+
+				if err := c.ClusterService.Update(cluster); err != nil {
+					return errors.WithMessagef(err, "failed to reconcile clientID of waiting for Kas Fleetshard Operator cluster %s: %s", cluster.ClusterID, err.Error())
+				}
+			}
+		}
+	}
```

Comment:
This code looks very similar to https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/153f51a85c1ce5d768439594d84aaa86da323a61/internal/kafka/internal/workers/clusters_mgr.go#L461-L477 except for the log messages. 

do you think it is something that can be extracted into an method to reduce duplication?

Reply:
Yea, I think that's a good idea :)